File: examples/access/constraints/range_basic.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Create a range constraint capping the maximum slant range to a satellite.
"""

import brahe as bh

# Require satellite slant range to be no more than 5000 km
constraint = bh.RangeConstraint(min_range_m=None, max_range_m=5_000_000.0)

print(f"Created: {constraint}")

File: examples/access/constraints/range_basic.rs
//! Create a range constraint capping the maximum slant range to a satellite.

#[allow(unused_imports)]
use brahe as bh;

fn main() {
    // Require satellite slant range to be no more than 5000 km
    let constraint = bh::RangeConstraint::new(None, Some(5_000_000.0)).unwrap();

    println!("Created: {}", constraint);
}
